### ISSUE TITLE
Shutdown after NPE while join stuck

### ISSUE THREAD

**Issue by cstamas** [hazelcast/hazelcast#8734]
Followup of https://github.com/hazelcast/hazelcast/issues/8733

While shutting `nodeA` (see https://github.com/hazelcast/hazelcast/issues/8733) after `nodeB` joined, `nodeA` does not shut down, instead logs this:

```
jvm 1    | 2016-08-17 12:29:19,997+0200 INFO  [WrapperListener_stop_runner] *SYSTEM com.hazelcast.core.LifecycleService - [127.0.0.1]:5701 [hzgroup] [3.6.4] Address[127.0.0.1]:5701 is SHUTTING_DOWN
jvm 1    | 2016-08-17 12:29:20,001+0200 INFO  [WrapperListener_stop_runner] *SYSTEM com.hazelcast.partition.InternalPartitionService - [127.0.0.1]:5701 [hzgroup] [3.6.4] Some backup replicas are inconsistent with primary, waiting for synchronization. Timeout: 599998ms
jvm 1    | 2016-08-17 12:29:21,005+0200 INFO  [WrapperListener_stop_runner] *SYSTEM com.hazelcast.partition.InternalPartitionService - [127.0.0.1]:5701 [hzgroup] [3.6.4] Some backup replicas are inconsistent with primary, waiting for synchronization. Timeout: 598997ms
jvm 1    | 2016-08-17 12:29:22,012+0200 INFO  [WrapperListener_stop_runner] *SYSTEM com.hazelcast.partition.InternalPartitionService - [127.0.0.1]:5701 [hzgroup] [3.6.4] Some backup replicas are inconsistent with primary, waiting for synchronization. Timeout: 597996ms
jvm 1    | 2016-08-17 12:29:23,017+0200 INFO  [WrapperListener_stop_runner] *SYSTEM com.hazelcast.partition.InternalPartitionService - [127.0.0.1]:5701 [hzgroup] [3.6.4] Some backup replicas are inconsistent with primary, waiting for synchronization. Timeout: 596995ms
jvm 1    | 2016-08-17 12:29:24,022+0200 INFO  [WrapperListener_stop_runner] *SYSTEM com.hazelcast.partition.InternalPartitionService - [127.0.0.1]:5701 [hzgroup] [3.6.4] Some backup replicas are inconsistent with primary, waiting for synchronization. Timeout: 595993ms
jvm 1    | 2016-08-17 12:29:25,027+0200 INFO  [WrapperListener_stop_runner] *SYSTEM com.hazelcast.partition.InternalPartitionService - [127.0.0.1]:5701 [hzgroup] [3.6.4] Some backup replicas are inconsistent with primary, waiting for synchronization. Timeout: 594992ms
...
```

Most probably caused by https://github.com/hazelcast/hazelcast/issues/8733

Hazelcast version: 3.6.4
Java: Oracle Java8
OS: dev OSX, prod Linux

**Comment by jerrinot** [comment]
@cstamas: thanks for another useful report. did the `nodeA` recover/shutdown eventually?

**Comment by cstamas** [comment]
@jerrinot did not wait for timeout of 599998ms :) I killed it. Subsequent reboot/join went fine. Most probably the https://github.com/hazelcast/hazelcast/issues/8733 messed it up.

**Comment by jerrinot** [comment]
ok. timeout is an upper bound. there is a chance it would recover before the dead-line.

**Comment by jerrinot** [comment]
this issue has the same root cause as #8733. It's fixed in 3.7
